webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/nodes/Properties.scala

```diff
@@ -233,11 +233,10 @@ object GenericProperty {
    */
   def parseValue(value: String): PureResult[ConfigValue] = {
```

Comment:
Why is it parsed from a string? We have to chose a type when creating properties, can't we use this information?

Reply:
not really - at least, not in 6.1 without changing *a lot* of things in backend. But even if we try to take advantage of that, it means that we need to tag that type everywhere a parameter can come from, including:

- from API 
- from archive of parameters serialized before we were able to specify types 
- from ldap.

And since we have the same format for nodes (and group) which we stored at time when parameters where just string, we can't just assume "string" for already saved values.

Finally, backend parsing will likelly remains more consistent than front-end, since front end doesn't know what hocon is (ie: we will likelly need to take shortcut in UI to change values to json and only check superficially, and then let backend really parsed value and decide)

Reply:
Ok!